### PR TITLE
ChunkGeneratorConfig minY and maxY to getTopBedrockY and getBottomBedrockY

### PR DESCRIPTION
**Pull request by falseresync**

### CODE REVIEW

**Comment by Juuxel** [comment]
These are overridden in `CavesChunkGeneratorConfig` to return 0 for min Y and 127 for max Y 🤔

**Comment by falseresync** [comment]
Lol wut

**Comment by falseresync** [comment]
Hmm `getMaxY = 0` in `OverworldChunkGeneratorConfig` too. That is super weird

**Comment by falseresync** [comment]
But my way seems correct (?) In `SurfaceChunkGenerator.buildBedrock` they're not used as in my variant

**Comment by Juuxel** [comment]
Could you update the base branch to 20w10a?

**Comment by Runemoro** [comment]
@falseresync I don't see the problem, they're not swapped...

**Comment by liach** [comment]
So this one actually relates to #1123 and are both troubled by this min/max y thing

**Comment by falseresync** [comment]
They're probably not `min`/`max` but rather some sort of `limit`, which can be either. But there're 2 of them. Maybe one should be `limitY` (or `yLimit`?) and the other one is `min` or `max`?

**Comment by liach** [comment]
How are these 2 fields used, just curious?

**Comment by falseresync** [comment]
They’re used as min and max respectively, but they switch their roles in 2(?) cases

**Comment by liach** [comment]
> They’re used as min and max respectively, but they switch their roles in 2(?) cases

Like how? in a loop? Show the actual usage in enigma/decompiled code

**Comment by falseresync** [comment]
I think they’re used in 1 place only, and switch their roles in children classes, aka not on the access side, but on the declaration side

**Comment by falseresync** [comment]
Cannot check rn! Will update when I can, anyone else does that

File: mappings/net/minecraft/world/gen/chunk/ChunkGeneratorConfig.mapping [hunk 1]
```diff
@@ -41,5 +41,5 @@ CLASS net/minecraft/class_2888 net/minecraft/world/gen/chunk/ChunkGeneratorConfi
 		ARG 1 state
 	METHOD method_12572 setDefaultFluid (Lnet/minecraft/class_2680;)V
 		ARG 1 state
-	METHOD method_16400 getMaxY ()I
-	METHOD method_16401 getMinY ()I
+	METHOD method_16400 getMinY ()I
+	METHOD method_16401 getMaxY ()I
```
Comment (liach) [inline mappings/net/minecraft/world/gen/chunk/ChunkGeneratorConfig.mapping]:
```suggestion
	METHOD method_16400 getTopBedrockY ()I
	METHOD method_16401 getBottomBedrockY ()I
```

**Review by liach — CHANGES_REQUESTED** [review]

**Review by liach — APPROVED** [review]

**Comment by Runemoro** [comment]
So what are the values of these for the overworld, where there is no top bedrock, and end, where there is no bedrock at all?

**Comment by liach** [comment]
0 top bedrock or 256 bottom bedrock would skip bottom/top bedrock generation.

**Review by jaskarth — APPROVED** [review]

**Comment by falseresync** [comment]
See #1157

**Pull request closed without merging**